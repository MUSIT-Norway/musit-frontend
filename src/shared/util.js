/* @flow */
import React from 'react';
import moment from 'moment';

export const flatten = (arr: []) => {
  const obj = {};

  for (let i = 0; i < arr.length; i++) {
    Object.keys(arr[i]).forEach((x) => {
      obj[x] = arr[i][x];
    });
  }

  return obj;
};

export const filter = (arr: any[], fields: string[], pattern: string) => {
  const contains = (s: any, p: string) => {
    return (s || '').toString().toLowerCase().indexOf(p.toLowerCase()) !== -1;
  };
  return arr.filter((row) => {
    return fields.find((field) => contains(row[field], pattern));
  });
};

export const getDisplayName = (Component: React.Component<*, *, *>) => {
  return Component.displayName || Component.name || 'Component';
};

export const blur = () => {
  // Give the document focus
  window.focus();

  // Remove focus from any focused element
  if (document.activeElement) {
    document.activeElement.blur();
  }
};

export const containsObjectWithField = (arr: any[], field: string, value: string): boolean => arr.filter((e) => e[field] === value).length > 0;

export const DATE_FORMAT_DISPLAY = 'DD.MM.YYYY';

export const parseUTCDate = (dateStr: string) => {
  return moment.utc(dateStr);
};

export const parseISODate = (dateStr: string) => {
  return moment(new Date(dateStr));
};
export const formatISOString = (d: Date)=> {
  return moment(d).format('YYYY-MM-DDT00:00:00.000Z');
};

export const parseFloatFromString = (value: string): number => {
  return typeof value === 'string' ? window.parseFloat(value.replace(',', '.')) : value;
};

export const formatFloatToString = (number: number): string => {
  return typeof number === 'number' ? number.toString().replace('.', ',') : number;
};

export const hasProp = (obj: any, prop: string): boolean => {
  return {}.hasOwnProperty.call(obj, prop);
};


export const customSortingStorageNodeType = (type: string): string => {
  switch (type) {
  case 'Organisation' : return '01';
  case 'Building': return '02';
  case 'Room': return '03';
  case 'StorageUnit': return '04';
  default: return '99';
  }
};

export const isDateBiggerThanToday = (newDate: any): boolean => {
  const today = moment();
  const isAfterYear = moment(newDate).isAfter(today, 'year');
  const isAfterMonth = moment(newDate).isAfter(today, 'month');
  const isAfterDay = moment(newDate).isAfter(today, 'day');
  return isAfterDay || isAfterMonth || isAfterYear;
};

export class Option {
  value = null;

  constructor(value: any) {
    this.value = value;
  }

  map<U>(func: (a: any) => U): ?U {
    if (this.value == null) {
      return; // return undefined/void
    }
    return func(this.value);
  }

}

const testing = process.env.NODE_ENV === 'test';

export const apiUrl = (url: string): string => {
  return `${testing ? 'http://localhost' : ''}${url}`;
};

export const getPath = (node: any) => {
  const nodeIds = (node.path != null ? node.path : '').split(',').slice(1).map(p => parseFloat(p)).filter(n => n);
  const pathNames = node.pathNames || [{
    nodeId: node.id,
    name: node.name
  }];
  return nodeIds.map(nodeId => {
    let pathMatch = pathNames.find(e => e.nodeId === nodeId);
    if (!pathMatch) {
      pathMatch = {
        nodeId: node.id,
        name: node.name
      };
    }
    return {
      id: pathMatch.nodeId,
      name: pathMatch.name,
      url: '/magasin/' + pathMatch.nodeId
    };
  });
};