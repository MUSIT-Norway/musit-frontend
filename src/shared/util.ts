// @flow
import * as React from 'react';
import * as moment from 'moment';
import { Observable } from 'rxjs';
import { PathName } from '../types/node';
import { AppSession } from '../types/appSession';
import { Star, Maybe, MUSTFIX } from '../types/common';

export const toPromise = (fn: (val: Star) => Observable<Star>) => (val: Star) =>
  fn(val).toPromise();

export const flatten = (arr: Array<Star>) => {
  const obj = {};

  for (let i = 0; i < arr.length; i++) {
    Object.keys(arr[i]).forEach(x => {
      obj[x] = arr[i][x];
    });
  }

  return obj;
};

export const filter = (arr: Array<Star>, fields: Array<string>, pattern: string) => {
  const contains = (s: Star, p: string) => {
    return (
      (s || '')
        .toString()
        .toLowerCase()
        .indexOf(p.toLowerCase()) !== -1
    );
  };
  return arr.filter(row => {
    return fields.find(field => contains(row[field], pattern));
  });
};

export const getDisplayName = (Component: React.SFC<any>) => {
  return Component.displayName || Component.name || 'Component';
};
/* I changed the above to SFC instead of Component, as Component isn't supposed to have displayname nor name.
Do something else if non-SFCs are used as argument to the above function.

export const getDisplayName = (Component: React.Component<Star, Star>) => {
  return Component.displayName || Component.name || 'Component';
};

*/
export const blur = () => {
  // Give the document focus
  window.focus();

  // Remove focus from any focused element
  if (document.activeElement) {
    (document.activeElement as HTMLElement).blur();
  }
};

export const containsObjectWithField = (
  arr: Array<Star>,
  field: string,
  value: string
): boolean => arr.filter(e => e[field] === value).length > 0;

export const DATE_FORMAT_DISPLAY = 'DD.MM.YYYY';

export const parseUTCDate = (dateStr: string) => {
  return moment.utc(dateStr);
};

export const parseISODate = (dateStr: string) => {
  return moment(new Date(dateStr));
};
export const formatISOString = (d: Date | string) => {
  return moment(d).format('YYYY-MM-DDT00:00:00.000Z');
};

export const maybeFormatISOString = (d?: Date | string) => {
  return d ? formatISOString(d) : d;
  // moment(d).format('YYYY-MM-DDT00:00:00.000Z');
};

export const parseFloatFromString = (value: string): number => {
  return typeof value === 'string' ? parseFloat(value.replace(',', '.')) : value;
};

export const formatFloatToString = (number: number): string => {
  return typeof number === 'number' ? number.toString().replace('.', ',') : number;
};

export const hasProp = (obj: Star, prop: string): boolean => {
  return {}.hasOwnProperty.call(obj, prop);
};

export const customSortingStorageNodeType = (type: string): string => {
  switch (type) {
    case 'Organisation':
      return '01';
    case 'Building':
      return '02';
    case 'Room':
      return '03';
    case 'StorageUnit':
      return '04';
    default:
      return '99';
  }
};

export const isDateBiggerThanToday = (newDate: string | number): boolean => {
  const today = moment();
  const isAfterYear = moment(newDate).isAfter(today, 'year');
  const isAfterMonth = moment(newDate).isAfter(today, 'month');
  const isAfterDay = moment(newDate).isAfter(today, 'day');
  return isAfterDay || isAfterMonth || isAfterYear;
};

export class Option {
  value = null;

  constructor(value: Star) {
    this.value = value;
  }

  map<U>(func: (a: Star) => U): Maybe<U> {
    if (this.value == null) {
      return; // return undefined/void
    }
    return func(this.value);
  }
}

const testing = process.env.NODE_ENV === 'test';

export const apiUrl = (url: string): string => {
  return `${testing ? 'http://localhost' : ''}${url}`;
};

export type SomethingLikeNode = {
  id: number;
  nodeId: string;
  name: string;
  path: string;
  pathNames: Array<PathName>;
};

export const getPath = (node: SomethingLikeNode) => {
  if (!node) {
    return [];
  }
  const nodeIds = (node.path || '')
    .split(',')
    .slice(1)
    .map(p => parseFloat(p))
    .filter(n => n);
  const pathNames = node.pathNames || [
    {
      nodeId: node.id,
      nodeUuid: node.nodeId,
      name: node.name
    }
  ];
  return nodeIds.map(nodeIntId => {
    let pathMatch = pathNames.find(e => e.nodeId === nodeIntId);
    if (!pathMatch) {
      pathMatch = {
        nodeId: node.nodeId as MUSTFIX, //TODO: The code is confusing regarding
        // whether nodeIs is a number or string. Please remove/fix this confusion!
        nodeUuid: node.nodeId,
        name: node.name
      };
    }
    return {
      id: pathMatch.nodeUuid,
      nodeId: pathMatch.nodeUuid,
      name: pathMatch.name,
      url: '/magasin/' + pathMatch.nodeUuid
    };
  });
};

export const musitParseFloat = (txt: string) => {
  if (!txt) return undefined;
  if (txt.length === 1) {
    if (isNaN(parseInt(txt))) return '';
    else return txt;
  }

  const res = txt.replace(',', '.');

  const reg = /(.*?)(0*)$/;
  const svar = reg.exec(res);
  if (svar) {
    const endNulls = svar[2];
    let startTxt = svar[1];
    const maybeDot = startTxt.endsWith('.') ? '.' : '';
    /*  console.log('svaret er : ' + svar); */
    if (startTxt) startTxt = parseFloat(startTxt).toString();

    return startTxt + maybeDot + endNulls;
  } else
    throw new Error(
      'Internal error, unexpected result on regular expression in musitParseFloat'
    );
};

export const musitParseInt = (txt: string) => {
  if (txt) {
    if (txt.length === 1) {
      if (isNaN(parseInt(txt))) return '';
      else return txt;
    } else {
      return parseInt(txt);
    }
  }
  return undefined;
};

// constant for eventTypeId

export const conservationProcessTypeId = 1;
export const treatmentTypeId = 2;
export const technicalDescriptionTypeId = 3;
export const storageAndHandlingTypeId = 4;
export const hseRiskAssessmentTypeId = 5;
export const conditionAssessmentTypeId = 6;
export const reportTypeId = 7;
export const materialDeterminationTypeId = 8;
export const measurementDeterminationTypeId = 9;
export const noteTypeId = 10;

// "collections"
export const algaeCollectionUuid = '1d8dd4e6-1527-439c-ac86-fc315e0ce852';
export const archaeologyCollectionUuid = '2e4f2455-1b3b-4a04-80a1-ba92715ff613';
export const entomologyCollectionUuid = 'ba3d4d30-810b-4c07-81b3-37751f2196f0';
export const ethnographyCollectionUuid = '88b35138-24b5-4e62-bae4-de80fae7df82';
export const vascularPlantsCollectionUuid = '7352794d-4973-447b-b84e-2635cafe910a';
export const lichensCollectionUuid = 'fcb4c598-8b05-4095-ac00-ce66247be38a';
export const marineInvertebratesCollectionUuid = 'ef4dc066-b6f8-4155-89f8-7aa9aeeb2dc4';
export const bryophyteCollectionUuid = 'd0dd5ad3-c22f-4ea0-8b52-dc5b0e17aa24';
export const numismaticCollectionUuid = '8bbdf9b3-56d1-479a-9509-2ea82842e8f8';
export const fungiCollectionUuid = '23ca0166-5f9e-44c2-ab0d-b4cdd704af07';
export const allCollectionUuid = '00000000-0000-0000-0000-000000000000';

export const getCultureOrNatureUnit = (appSession?: AppSession) => {
  if (appSession) {
    const collectionId = appSession.collectionId;
    const culture = [
      archaeologyCollectionUuid,
      ethnographyCollectionUuid,
      numismaticCollectionUuid
    ];
    if (culture.includes(collectionId)) return '(cm)';
    else return '(mm)';
  } else return '(undefined)';
};

export const styleWidth = (pixels: number) => ({ width: pixels });
export const styleWidth10 = styleWidth(10);

export const musitCoodinateValidate = (coordinateType?: string) => (value: string) => {
  if (coordinateType === 'MGRS') {
    const coorRegexMGRS1 = new RegExp(
      /^[A-Z]{2}(-[A-Z]{2})?\s((\d{1}(-\d{1})?,\d{1}(-\d{1})?)|(\d{2}(-\d{2})?,\d{2}(-\d{2})?)|(\d{3}(-\d{3})?,\d{3}(-\d{3})?)|(\d{4}(-\d{4})?,\d{4}(-\d{4})?)||(\d{5}(-\d{5})?,\d{5}(-\d{5})?))$/,
      'i'
    ); // LL-LK 234-345, 234-456
    //    const coorRegexMGRS2 = new RegExp(/^[A-Z]{2}(\d{2}|\d{4}|\d{6}|\d{8}|\d{10})$/, 'i'); //LL2345656789
    if (coorRegexMGRS1.test(value)) {
      return true;
    } else {
      return false;
    }
  }
  if (coordinateType === 'LAT/LONG') {
    const r1 = new RegExp(/^\d+(\s+\d+(\s+\d+)?)?(N|S)\s+\d+(\s+\d+(\s+\d+)?)?(E|W)$/);
    const r2 = new RegExp(/^\-?\d+(\.\d+)?\s+\-?\d+(\.\d+)?$/);
    return r1.test(value) || r2.test(value);
  }

  if (coordinateType === 'UTM') {
    const r1 = new RegExp(/^\d+(\s+|\,)\d+$/);
    return r1.test(value);
  }
  return true;
};

export const validatePositiveInteger = (count: string) => {
  const intRegex = new RegExp(/^\d+$/);
  if (count === '') {
    return true;
  } else {
    if (intRegex.test(count)) {
      return true;
    } else {
      return false;
    }
  }
};
