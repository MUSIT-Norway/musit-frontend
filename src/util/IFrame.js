import React from 'react';
import ReactDOM from 'react-dom';

class IFrame extends React.Component {

  static propTypes = {
    frameProps: React.PropTypes.object,
    content: React.PropTypes.string.isRequired,
    onLoad: React.PropTypes.func
  };

  static defaultProps = {
    frameProps: {
      frameBorder: 0
    }
  };

  constructor(props) {
    super(props);
    this.onLoad = this.onLoad.bind(this);
  }

  updateIFrameContents() {
    const contentWindow = this.domNode.contentWindow;
    contentWindow.document.open();
    contentWindow.document.write(this.props.content);
    contentWindow.document.close();
  }

  onLoad() {
    if (typeof this.props.onLoad === 'function') {
      this.props.onLoad(this.domNode);
    }
  }

  render() {
    return (
      <iframe {...this.props.frameProps} onLoad={this.onLoad} />
    );
  }

  componentDidMount() {
    this.domNode = ReactDOM.findDOMNode(this);
    this.updateIFrameContents();
  }

  componentDidUpdate() {
    this.updateIFrameContents();
  }
}

export default IFrame;