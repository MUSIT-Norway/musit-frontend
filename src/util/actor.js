import find from 'lodash/find';

export const getActorId = (actor) => {
  return actor.dataportenId || actor.applicationId;
};

export const getActorNames = (actors, doneById, registeredById) => {
  const doneBy = find(actors, (a) => a.dataportenId === doneById || a.applicationId === doneById);
  const registeredBy = find(actors, (a) => a.dataportenId === registeredById || a.applicationId === registeredById);
  return {
    doneBy: doneBy && doneBy.fn,
    registeredBy: registeredBy && registeredBy.fn
  };
};