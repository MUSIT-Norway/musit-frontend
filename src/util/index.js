/* @flow */
import moment from 'moment'

export const flatten = (arr: []) => {
  const obj = {};

  for (let i = 0; i < arr.length; i++) {
    Object.keys(arr[i]).forEach((x) => {
      obj[x] = arr[i][x]
    })
  }

  return obj
}

export const blur = () => {
  // Give the document focus
  window.focus();

  // Remove focus from any focused element
  if (document.activeElement) {
    document.activeElement.blur();
  }
}

class PathName {
  nodeId: number;
  name: string;

  constructor(nodeId: number, name: string) {
    this.nodeId = nodeId
    this.name = name;
  }
}

class BreadCrumb {
  id: number;
  name: string;
  url: string;

  constructor(pathName: ?PathName) {
    if (pathName) {
      this.id = pathName.nodeId;
      this.name = pathName.name;
      this.url = '/magasin/' + pathName.nodeId;
    }
  }
}

export const createBreadcrumbPath = (pathStr: string, pathNames: PathName[]): BreadCrumb[]  => {
  const pathStrIds = (pathStr != null ? pathStr : '').slice(1, -1).split(',').slice(1).map(p => parseFloat(p));
  return pathStrIds.map(pathId => new BreadCrumb(pathNames.find(e => e.nodeId === pathId)))
}

export const containsObjectWithField = (arr: any[], field: string, value: string): boolean => arr.filter((e) => e[field] === value).length > 0

export const DATE_FORMAT_DISPLAY = 'DD.MM.YYYY'
export const DATE_FORMAT_ISO_SHORT = 'YYYY-MM-DD'
export const DATE_FORMAT_ISO_FULL = 'YYYY-MM-DDTHH:mm:ss.SSSZZ'

export const parseISODateNonStrict = (dateStr: string) => {
  return moment(dateStr, [DATE_FORMAT_ISO_SHORT])
}

export const parseISODateStrict = (dateStr: string) => {
  return moment(dateStr, [DATE_FORMAT_ISO_SHORT], true)
}

export const parseFloatFromString = (value: string): number => {
  return typeof value === 'string' ? window.parseFloat(value.replace(',', '.')) : value
}

export const formatFloatToString = (number: number): string => {
  return typeof number === 'number' ? number.toString().replace('.', ',') : number
}

export const hasProp = (obj: any, prop: string): boolean => {
  return {}.hasOwnProperty.call(obj, prop)
}

export class Option {
  value = null;

  constructor(value: any) {
    this.value = value;
  }

  map<U>(func: (a: any) => U): ?U {
    if (this.value == null) {
      return; // return undefined/void
    }
    return func(this.value)
  }
}
