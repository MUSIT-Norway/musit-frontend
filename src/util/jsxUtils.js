import React from 'react'

export function wrapWithContext(childContextTypes, childContext, child){

  class WrapperWithContext extends React.Component {
    static childContextTypes = { ...childContextTypes, closeModal: React.PropTypes.func };
    getChildContext() { return { ...childContext, closeModal: () => true }; }
    render() {
      return this.props.children
    }
  }

  return (
    <WrapperWithContext>
      {child}
    </WrapperWithContext>
  );
}