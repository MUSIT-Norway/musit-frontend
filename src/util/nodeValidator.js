import { I18n } from 'react-i18nify';

export const getPathLength = (formProps) => {
  const { pathNames } = formProps || {};
  return pathNames && pathNames.length;
};

export const checkNodeType = (from, to) => {

  const matchFromType = from.value ? from.value.type : from.type;

  if (to.type === 'Root' && 'Organisation' !== matchFromType) {
    return I18n.t('musit.storageUnits.type.organisationAllowedToMove');
  }

  if (2 === getPathLength(to) && to.type === 'Organisation' && 'Building' !== matchFromType) {
    return I18n.t('musit.storageUnits.type.buildingAllowedToMove');
  }

};

export const checkNodeBranch = (from, to) => {

  const pathNameFound = to.pathNames ? to.pathNames.filter(id => {
    const matchOnValue = from.value && id.nodeId === from.value.id;
    return id.nodeId === from.id || matchOnValue;
  }) : {};

  if ( pathNameFound.length > 0) {
    return I18n.t('musit.storageUnits.type.notAllowedToMove');
  }
};

export const checkNodeBranchAndType = (from, to) => {
  // console.log(JSON.stringify(from));
  // console.log(JSON.stringify(to));
  // console.log('checkNodeBranchAndType return =' + checkNodeType(from, to) + checkNodeBranch(from, to));

  return checkNodeType(from, to) || checkNodeBranch(from, to);
};

