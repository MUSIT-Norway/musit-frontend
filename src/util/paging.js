import React, { PropTypes } from 'react';
import { range } from 'lodash';

class PagingToolbar extends React.Component {
  render() {
    const numPages = this.props.numItems / this.props.perPage;
    const currentPage = this.props.currentPage;
    const maxPages = 10;
    const min = currentPage;
    const max = maxPages;

    return (
      <div
        style={{
          float: 'right',
          marginRight: '30px'
        }}
      >
      <span
        style={{
          fontWeight: 'bold'
        }}
      >
        {numPages > 1 && min > 1 ?
          <a
            href="/page/back"
            onClick={(e) => {
              e.preventDefault();
              this.props.onClick(currentPage - 1);
            }}
          >
            {'<'}
          </a>
          : '<'
        }
      </span>
        {min > 1
            ?
            <a
              href={`/page/${min}`}
              onClick={(e) => {
                e.preventDefault();
                this.props.onClick(min - maxPages > 1 ? min : 1);
              }
              }
            >
            {' .. '}
            </a> : null
            }
        {range(min, max).map((page, i) => {
          return (
            <span
              key={i}
              style={{
                padding: '5px'
              }}
            >
            {currentPage === page ?
              page
              :
              <a
                href={`/page/${page}`}
                onClick={(e) => {
                  e.preventDefault();
                  this.props.onClick(page);
                }}
              >
                {page}
              </a>
            }
          </span>
          );
        })}
        <span
          style={{
            fontWeight: 'bold'
          }}
        >
    {numPages - min > maxPages
      ?
      <a
        href={`/page/${min}`}
        onClick={(e) => {
          e.preventDefault();
          this.props.onClick(min);
        }
        }
      >
        {' .. '}
      </a> : null
    }
        {numPages > 1 && currentPage < numPages ?
          <a
            href="/page/next"
            onClick={(e) => {
              e.preventDefault();
              this.props.onClick(currentPage + 1);
            }}
          >
            {'>'}
          </a>
          : '>'
        }
      </span>
      </div>
    );
  }
}

PagingToolbar.propTypes = {
  currentPage: PropTypes.number.isRequired,
  numItems: PropTypes.number.isRequired,
  baseUrl: PropTypes.string.isRequired,
  perPage: PropTypes.number,
  onClick: PropTypes.func.isRequired
};

PagingToolbar.defaultProps = {
  perPage: 50
};

export default PagingToolbar;