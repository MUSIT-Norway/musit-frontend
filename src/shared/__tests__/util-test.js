import assert from 'assert';
import { isDateBiggerThanToday, parseUTCDate, customSortingStorageNodeType, localToISOString } from '../util';
import moment from 'moment';

describe('parseUTCDate', () => {
  it('should accept full iso timestamp', () => {
    const date = new Date('2016-09-07T00:56:00.000Z');
    const parsed = parseUTCDate(date);
    expect(parsed).toMatchSnapshot();
  });

  it('should accept standard simple ISO date format', () => {
    const date = '2016-12-23';
    const parsed = parseUTCDate(date);
    assert(parsed.isValid() === true);
  });
});

describe('customSortingStorageNodeType', () => {
  it('Order should be Organisation < Building < Room < StorageUnit', () => {
    const unknown = customSortingStorageNodeType('errrr');
    const building = customSortingStorageNodeType('Building');
    const organisation = customSortingStorageNodeType('Organisation');
    const storageUnit = customSortingStorageNodeType('StorageUnit');
    const room = customSortingStorageNodeType('Room');
    assert(organisation < building && building < room && room < storageUnit && storageUnit < unknown);
  });

});


describe('isDateBiggerThanToday', () => {
  it('Check Today', () => {
    const newDate = moment();
    assert(isDateBiggerThanToday(newDate) === false);
  });
  it('Check one day plus Today', () => {
    const newDate = moment().add(1, 'd');
    assert(isDateBiggerThanToday(newDate) === true);
  });
  it('Check one month plus Today', () => {
    const newDate = moment().add(1, 'M');
    assert(isDateBiggerThanToday(newDate) === true);
  });
  it('Check one year plus Today', () => {
    const newDate = moment().add(1, 'y');
    assert(isDateBiggerThanToday(newDate) === true);
  });
  it('Check one day subtract from Today', () => {
    const newDate = moment().subtract(1, 'd');
    assert(isDateBiggerThanToday(newDate) === false);
  });
  it('Check one month subtract from Today', () => {
    const newDate = moment().subtract(2, 'M');
    assert(isDateBiggerThanToday(newDate) === false);
  });
  it('Check one year subtract from Today', () => {
    const newDate = moment().subtract(1, 'y');
    assert(isDateBiggerThanToday(newDate) === false);
  });
  it('Check Today minus one month three days', () => {
    let newDate = moment().subtract(3, 'd');
    newDate = newDate.subtract(1, 'd');
    assert(isDateBiggerThanToday(newDate) === false);
  });
  it('Check Today minus one month 15 days', () => {
    let newDate = moment().subtract(15, 'd');
    newDate = newDate.subtract(1, 'd');
    assert(isDateBiggerThanToday(newDate) === false);
  });
  it('Check Today minus one month 29 days', () => {
    let newDate = moment().subtract(29, 'd');
    newDate = newDate.subtract(1, 'd');
    assert(isDateBiggerThanToday(newDate) === false);
  });
  it('Check Today plus one month 14 days', () => {
    let newDate = moment().add(14, 'd');
    newDate = newDate.add(1, 'd');
    assert(isDateBiggerThanToday(newDate) === true);
  });
  it('Check Today plus one month 29 days', () => {
    let newDate = moment().add(29, 'd');
    newDate = newDate.add(1, 'd');
    assert(isDateBiggerThanToday(newDate) === true);
  });
  it('Check blank', () => {
    const newDate = '';
    assert(isDateBiggerThanToday(newDate) === false);
  });
  it('Check null', () => {
    const newDate = null;
    assert(isDateBiggerThanToday(newDate) === false);
  });
  it('Check undefined', () => {
    const newDate = undefined;
    assert(isDateBiggerThanToday(newDate) === false);
  });
});



describe('localToISOString', () => {

  it('localTOIsoString should always be equal to input date', () => {
    const newDate = new Date();
    const isoDat= localToISOString(newDate);
    assert(moment(newDate).format('YYYYMMDD') === moment(isoDat).format('YYYYMMDD'));
  });

  it('CheckAroundMidnight', () => {
    const newDate = new Date('2017-02-20T00:00:01');
    const isoDate= localToISOString(newDate);
    assert(moment(newDate).format('YYYYMMDD') === moment(isoDate).format('YYYYMMDD'));
  });

});


describe('toISOString', () => {

  it('Date local time should not be equal to isoDate if local time has timeZoneOffset <> 0', () => {
    const d = new Date();
    const momentLocal = moment(d); // Local date
    const i = momentLocal.toISOString().split('.')[0]; // Zulu time (without milliseconds)
    const l = momentLocal.format('YYYY-MM-DDTHH:mm:ssZ');
    assert(d.getTimezoneOffset() !== 0 || i !== l); // If timeZone is different from zulu, local should be different from isoString
  });
});

describe('parseUTC around midnight', () => {

  it('It shows wrong local date around midnight if local zone is GMT+1', () => {
    const str = '2017-01-12T00:01:00.000+01:00';
    const d = new Date(str);
    const m = moment(d).format('YYYYMMDD');
    const p = parseUTCDate(str).format('YYYYMMDD');
    assert(m !== p );
  });
});
