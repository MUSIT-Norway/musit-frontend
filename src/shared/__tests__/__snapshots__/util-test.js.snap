exports[`parseUTCDate should accept full iso timestamp 1`] = `"2016-09-07T00:56:00.000Z"`;
